### PR TITLE
Support Fargate

### PR DESCRIPTION
**Pull request by jonnyshaw89**
Allow containers to be run on either EC2 or Fargate

Fargate allows us to run containers serverless.

https://aws.amazon.com/fargate/

### CODE REVIEW

File: global_vars.tf [hunk 1]
```diff
@@ -51,6 +63,21 @@ variable "dns_record_name" {
 }
 
 # ECS
+variable "launch_type" {
+  description = "Where to launch the container"
+  default     = "EC2"
```
Comment (bitdivision) [inline global_vars.tf:68]: Worth changing description to include `FARGATE` / `EC2` options?

File: global_vars.tf [hunk 2]
```diff
@@ -122,5 +149,5 @@ variable "low_cpu_threshold" {
 
 variable "healthcheck_grace_period_seconds" {
   description = "Number of seconds to wait before first health check."
-  default     = 5
+  default     = 60
```
Comment (bitdivision) [inline global_vars.tf]: Fargate takes longer to startup I guess?

**Comment by bitdivision** [comment]
I'll see if I can do some testing.

File: global_vars.tf (hunk 2 above)
Comment (jonnyshaw89) [reply global_vars.tf]: Possibly not needed, ill test

Comment (jonnyshaw89) [reply global_vars.tf]: Not needed to be increased. I think this was added when the Author container used to take ages to start

Comment (samiwel) [reply global_vars.tf]: Start up should be pretty fast now, but building the container image takes longer instead. 🤦‍♂️ 	Still, I prefer the faster runtime over slower build time.

File: autoscaling.tf [hunk 3]
```diff
@@ -1,7 +1,11 @@
 resource "aws_appautoscaling_target" "ecs_service_target" {
+  depends_on = [
+    "aws_ecs_service.service",
+    "aws_ecs_service.fargate_service"
+  ]
```
Comment (samiwel) [inline autoscaling.tf:5]: Am I correct in thinking that the explicit dependency is only ever on one of these resources? Or is there a scenario where the EC2 and FARGATE ecs services are both depended upon?

**Review by samiwel — APPROVED** [review]

File: autoscaling.tf (hunk 3 above)
Comment (jonnyshaw89) [reply autoscaling.tf:5]: yeh 1 of

**Review by bitdivision — APPROVED** [review]
Tested runner under fargate, seems to work. Changes all look good to me.

**Pull request merged**